examples/generative/corrdiff/generate.py

```diff
@@ -378,6 +379,17 @@ def generate_and_save(dataset, sampler, f: nc.Dataset, generate_fn, device, logg
     time_index = -1
     writer = writer_from_input_dataset(f, dataset)
 
+    # Initialize threadpool for writers
+    writer_executor = ThreadPoolExecutor(max_workers=2)
```

Comment:
Why do you set the `num_workers` to `2`? Shouldn't this be configurable?

Reply:
Updating this to be a parameter in the configuration.